edb/server/compiler/compiler.py

```diff
@@ -1557,6 +1559,72 @@ def _compile_ql_query(
     is_explain = explain_data is not None
     current_tx = ctx.state.current_tx()
 
+    # Embed the normalized EdgeQL query info in the SQL as a comment.
+    # It has 2 layers of embedded JSON:
+    #   1. sql_info: the outer comment consumed by the edb_stat_statements
+    #      extension, used to replace the compiled SQL with the frontend query
+    #      in this JSON, as well as the `queryId` (unique index of statistics).
+    #   2. query_info: the inner comment that is stored in pg_stat_statements,
+    #      used as a supplement storage of the additional query information,
+    #      like query_type, output_format, etc.
+    sql_info: Dict[str, Any] = {}
+    if (
+        not ctx.bootstrap_mode
+        and ctx.backend_runtime_params.has_stat_statements
+    ):
+        user_schema = current_tx.get_user_schema()
+        sql_info['query'] = qlcodegen.generate_source(ql)
+        spec = ctx.compiler_state.config_spec
+        query_info: Dict[str, Any] = {
+            'type': 'EdgeQL',
+            # compilation_config
+            'cc': config.to_json_obj(
+                spec,
+                {
+                    **current_tx.get_system_config(),
+                    **current_tx.get_database_config(),
+                    **current_tx.get_session_config(),
+                },
+                setting_filter=lambda v: v.name in spec
+                                         and spec[v.name].affects_compilation,
```

Comment:
Can we make sure nothing `secret` goes in also?

Reply:
I think the only compilation setting that may include free text is `force_database_error`, which shouldn't have secrets. I can also add a double-safe to filter out extension configs in #7869.